extensions.go

```diff
@@ -624,3 +624,99 @@ func (c CookieExtension) Marshal() ([]byte, error) {
 func (c *CookieExtension) Unmarshal(data []byte) (int, error) {
 	return syntax.Unmarshal(data, c)
 }
+
+// struct {
+//         select(ClientOrServerExtension) {
+//             case client:
+//               CertificateType client_certificate_types<1..2^8-1>;
+//             case server:
+//               CertificateType client_certificate_type;
+//         }
+// } ClientCertTypeExtension;
+type ClientCertTypeExtension struct {
+	HandshakeType    HandshakeType     `tls:"none"`
+	CertificateTypes []CertificateType `tls:"head=1,min=1"`
```

Comment:
I haven't checked, but won't this effectively cause an assert here if you set the settings to not allow any types.

Reply:
Correct, but that's what the RFC specifies.

Reply:
Right I'm saying you might want this earlier up.